client/app/shared/problemDataService.js

```diff
@@ -0,0 +1,71 @@
+angular.module('RecallJS')
+  .factory('ProblemData', ProblemData);
+
+function ProblemData($http, $window){
+
+  return {
```

Comment:
I'm wondering if we should also include a property called problems (or something), that way we only have to do the GET request once (when the user successfully logs in, as opposed to having to call it multiple times whenever we need an array of problems). What do you think?